src/rgw/rgw_rest_s3.cc

```diff
@@ -3513,38 +3535,37 @@ int RGWRestoreObj_ObjStore_S3::get_params(optional_yield y)
 
 void RGWRestoreObj_ObjStore_S3::send_response()
 {
-  if (op_ret < 0)
-  {
-    set_req_state_err(s, op_ret);
+   
+  auto attr_iter = attrs.find(RGW_ATTR_RESTORE_STATUS);
+
+  if (restore_op < 0) {
+    set_req_state_err(s, restore_op);
     dump_errno(s);
     end_header(s, this);
     dump_start(s);
     return;
   }
 
-  rgw::sal::Attrs attrs = s->object->get_attrs();
-  auto attr_iter = attrs.find(RGW_ATTR_RESTORE_STATUS);
-  rgw::sal::RGWRestoreStatus restore_status;
-  if (attr_iter != attrs.end()) {
-    bufferlist bl = attr_iter->second;
-    auto iter = bl.cbegin();
-    decode(restore_status, iter);
-  }
-  ldpp_dout(this, 10) << "restore_status=" << restore_status << dendl;
-  
-  if (attr_iter == attrs.end() || restore_status != rgw::sal::RGWRestoreStatus::None) {
-    s->err.http_ret = 202; //Accepted
-    dump_header(s, "x-amz-restore", rgw_bl_str(restore_status));
-  } else if (restore_status != rgw::sal::RGWRestoreStatus::RestoreAlreadyInProgress) {
+  if (attr_iter == attrs.end() || restore_op == 0) {
+    if (rt == rgw::sal::RGWRestoreType::Temporary) {
```

Comment:
this if loop is not needed if restore_op == 0 right? And also why "attr_iter == attrs.end()" in this check?

Reply:
As per https://docs.aws.amazon.com/AmazonS3/latest/userguide/restoring-objects.html , Restore: header should be included in response only if IN_PROGRESS or RESTORED.

Even sample response in https://docs.aws.amazon.com/AmazonS3/latest/API/API_RestoreObject.html for 202 Accepted does not contain Restore header

Reply:
Well I guess we had discussed that and you suggested to include it in accepted that time.

Reply:
Oh is it.. sorry I dint recall.. anyway please address it now.

Reply:
okay!